### PR TITLE
"Added transmission line module"

### PR DESCRIPTION
**Pull request by JMNietoatBusyBee**
"Transmission Line Module"

### CODE REVIEW

File: src/app/core/services/portal.service.ts [hunk 1]
```diff
@@ -322,4 +323,20 @@ getMunicipalities(): Observable<any[]> {
...
+getTransmissionProfiles(): Observable<any[]> {
+ return this._portalApi.getTransmissionProfiles();
+}
+
+getTransmissionProfile(id: number): Observable<ITransmissionProfile>{
+  return this._portalApi.getTransmissionProfile(id);
+}
+
+addTransmissionProfile(transmissionProfile: any): Observable<any> {
+  return this._portalApi.addTransmissionProfile(transmissionProfile);
```
Comment (vernienamca) [inline src/app/core/services/portal.service.ts:336]: Is this for saving of transmission profile. Can we name it to "createTransmissionProfile" instead?

File: src/app/core/services/portal.service.ts [hunk 2]
```diff
@@ -322,4 +323,20 @@ getMunicipalities(): Observable<any[]> {
...
+}
+
+updateTransmissionProfile(id: number, data: any): Observable<any> {
+  return this._portalApi.updateTransmissionProfile(id,data);
```
Comment (vernienamca) [inline src/app/core/services/portal.service.ts:340]: Be mindful on spacing between code please.

**Review by vernienamca — CHANGES_REQUESTED** [review]

File: src/app/core/services/portal.service.ts (hunk 1 above)
Comment (JMNietoatBusyBee) [reply src/app/core/services/portal.service.ts:336]: Applied.

File: src/app/pages/apps/transmission-line/transmission-line-list/transmission-line-list.component.html [hunk 3]
```diff
@@ -0,0 +1,115 @@
+<vex-page-layout>
+  <vex-page-layout-header class="pb-16 flex flex-col items-start justify-center">
+    <div [class.container]="layoutCtrl.value === 'boxed'" [class.px-gutter]="layoutCtrl.value === 'fullwidth'" 
+      class="w-full flex flex-col sm:flex-row justify-between">
+      <div>
+        <h1 class="title mt-0 mb-1">Transmission Line Profile({{ totalCount }})</h1>
+        <vex-breadcrumbs [crumbs]="['Profile Management', 'Transmission Line Profile']"></vex-breadcrumbs>      </div>
```
[the remark this answers is not in the document]
Comment (JMNietoatBusyBee) [reply src/app/pages/apps/transmission-line/transmission-line-list/transmission-line-list.component.html:7]: Applied.

File: src/app/pages/apps/transmission-line/transmission-line-routing.module.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+const routes: VexRoutes = [
+  {
+    path: '',
+    component: TransmissionLineListComponent,
+    data: {
+      toolbarShadowEnabled: false
+    }
+  },
+
```
[the remark this answers is not in the document]
Comment (JMNietoatBusyBee) [reply src/app/pages/apps/transmission-line/transmission-line-routing.module.ts:17]: Applied.

File: src/app/pages/apps/transmission-line/transmission-line.component.ts [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+  ngOnInit(): void {
+    combineLatest([
+      this.form.get('st').valueChanges,
+      this.form.get('sp').valueChanges,
+      this.form.get('cp').valueChanges,
+      this.form.get('wp').valueChanges,
+      this.form.get('slwt').valueChanges,
+    ])
+    .pipe(
+      takeUntil(this._onDestroy$)
```
[the remark this answers is not in the document]
Comment (JMNietoatBusyBee) [reply src/app/pages/apps/transmission-line/transmission-line.component.ts:141]: Applied.

File: src/app/pages/apps/transmission-line/transmission-line.component.ts [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+      if (!permission || permission.some(s => s.isRead) === false) {
+        this._router.navigate([`pages/error-401`]);
+      }
+      if(permission.some(s => s.isUpdate) === false) {
+        this.form.controls['lineStretch'].disable();
+        this.form.controls['voltageId'].disable();
+        this.form.controls['st'].disable();
+        this.form.controls['sp'].disable();
+        this.form.controls['cp'].disable();
+        this.form.controls['wp'].disable();
```
[the remark this answers is not in the document]
Comment (JMNietoatBusyBee) [reply src/app/pages/apps/transmission-line/transmission-line.component.ts:113]: Applied.

File: src/app/pages/apps/transmission-line/transmission-line.component.ts [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+    });
+  }
+
+  ngOnInit(): void {
+    combineLatest([
+      this.form.get('st').valueChanges,
+      this.form.get('sp').valueChanges,
+      this.form.get('cp').valueChanges,
+      this.form.get('wp').valueChanges,
+      this.form.get('slwt').valueChanges,
```
[the remark this answers is not in the document]
Comment (JMNietoatBusyBee) [reply src/app/pages/apps/transmission-line/transmission-line.component.ts:138]: Applied.

File: src/app/core/services/portal.service.ts (hunk 2 above)
Comment (JMNietoatBusyBee) [reply src/app/core/services/portal.service.ts:340]: Applied.

**Pull request merged**